wiring/inc/spark_wiring_cloud.h

```diff
@@ -365,7 +366,8 @@ class CloudClass {
 
     static bool connected(void) { return spark_cloud_flag_connected(); }
     static bool disconnected(void) { return !connected(); }
-    static void connect(void) {
+    static void connect(const spark::NetworkClass& network = spark::Network) {
+        spark_set_connection_property(SPARK_CLOUD_BIND_NETWORK_INTERFACE, static_cast<network_interface_t>(network), nullptr, nullptr);
```

Comment:
What should the behavior be when a user calls `Particle.connect(Cellular)` when cellular is off? 
Do we want to turn the hardware on? or keep the current behavior where Particle.connect()/disconnect() is decoupled from the network interface state?

Reply:
DIscussed this [here](https://s.slack.com/archives/CS0T32ZKK/p1700246458097789?thread_ts=1700241989.915009&cid=CS0T32ZKK), leaving this behavior as currently implemented